From ec06c7566574bd6dccba07398dbad43fa27045fa Mon Sep 17 00:00:00 2001
From: Ćukasz Patron <priv.luk@gmail.com>
Date: Thu, 22 May 2025 20:30:05 +0200
Subject: [PATCH] pyi_generator.py: Pass all args/kwargs to original formatannotation()

Python 3.14 changed formatannotation() function signature by adding `*,
quote_annotation_strings=True` at the end of it.

Since we only care about avoiding `text.removeprefix('typing.')`, let's
just pass extra args/kwargs as is to keep compatibility with < 3.14.

[1] https://github.com/python/cpython/pull/124669

Change-Id: Ic115ed5cc8766cdf60286e88f6cfa79e263f64c9
---

diff --git a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
index c5dc446..1031e0b 100644
--- a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
+++ b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/lib/pyi_generator.py
@@ -69,12 +69,12 @@
     backup = inspect.formatannotation
 
     @classmethod
-    def formatannotation(cls, annotation, base_module=None):
+    def formatannotation(cls, annotation, base_module=None, *args, **kwargs):
         if getattr(annotation, '__module__', None) == 'typing':
             # do not remove the prefix!
             return repr(annotation)
         # do the normal action.
-        return cls.backup(annotation, base_module)
+        return cls.backup(annotation, base_module, *args, **kwargs)
 
     @classmethod
     def fix_typing_prefix(cls, signature):
